src/game/logic/PlayerLogic.java

```diff
@@ -40,5 +47,19 @@ public boolean isPhaseOne(){
         return !isPhaseOneOver();
     }
 
+    public boolean move(int srcIndex, int destIndex) {
+        if(!gameBoard.getCell(srcIndex).isOccupiedBy(getActivePlayer()) || isPhaseOne())  //immediately return false if source index player token does not match or if phase one
+            return false;
+//        if(activePlayer.getPiecesLeft() > 3) {
+//          if (gameBoard.getCell(srcIndex).isAdjacentTo(destIndex))
+//              return gameBoard.moveFromTo(srcIndex, destIndex, getActivePlayer());
+//       }
+       else if(gameBoard.getCount(activePlayer.getPlayerToken()) <=3) {
```

Comment:
Go ahead an uncomment out these lines, I merged in the adjacencies to this. Then test of course